clang/test/CheckedCRewriter/fptrinstructcallermulti1.c

```diff
@@ -101,7 +101,7 @@ int *mul2(int *x) {
 //CHECK: _Ptr<int> mul2(_Ptr<int> x) { 
 
 struct fptr * sus(struct fptr *, struct fptr *);
-//CHECK: struct fptr * sus(struct fptr *, struct fptr *);
+//CHECK: struct fptr * sus(struct fptr *, struct fptr *y : itype(_Ptr<struct fptr>));
```

Comment:
Is this Correct? Why is the second parameter an itype? Can you point me to a place where we are passing a WILD pointer to second argument? Is this because of malloc?

Reply:
This is a multifile example and the other file is `fptrinstructcallermulti2.c`. This is where `sus` is defined. Looking at that file, it seems that it was previously being rewritten as `struct fptr * sus(struct fptr *x, struct fptr *y : itype(_Ptr<struct fptr>)) {` which is the same as what Aaron has changed it to, here. So this change makes the rewriting consistent, which it wasn't before.

However, I have the same question as you: Why are/were we rewriting to the `itype` in the first place? I don't see why that should happen.

Reply:
This is because of `malloc`:

```
#define size_t int
extern _Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);

struct fptr { 
    int *values; 
    char *name;
};

void foo() {
    struct fptr * x = malloc(sizeof(struct fptr));    
    int *y = malloc(sizeof(int));
}
```
Gets converted to:

```
#define size_t int
extern _Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);

struct fptr { 
    _Ptr<int> values; 
    _Ptr<char> name;
};

void foo() {
    struct fptr * x = malloc(sizeof(struct fptr));    
    int *y = malloc(sizeof(int));
}
```

Reply:
I have opened an another issue to keep track of this: https://github.com/plum-umd/checkedc-clang/issues/96, We can handle this in the other issue.